hil/client/base.py

```diff
@@ -60,9 +61,23 @@ def check_response(self, response):
             except ValueError:  # No JSON request body; typical
                                 # For methods PUT, POST, DELETE
                 return
-        else:
+        try:
             e = json.loads(response.content)
             raise FailedAPICallException(
                 error_type=e['type'],
                 message=e['msg'],
             )
+        # Catching 404's that do not return JSON
+        except ValueError:
+            return response.content
+
+    def find_reserved(self, string):
+        """Returns a list of illegal characters in a string"""
+        p = '[^A-Za-z0-9 \$\-\_\.\+\!\*\'\(\)\,]+'
```

Comment:
1. I don't think most of these need escaping when inside a character class. The dash can go at the end of the list to avoid it being interpreted as special. Escaping the single quote can be avoided by using double quotes to delmit the string as a whole.
2. Generally speaking, use `r"raw strings"` for regexes, so you don't have to worry about what needs to be escaped in a python string as well.

Reply:
Ok, I'll work on that.

Reply:
Thanks for the tip, https://github.com/CCI-MOC/hil/pull/926/commits/c1fea672dff1a55aa9504c82e45bde5d5bb90389 addresses this.